Jenkinsfile

```diff
@@ -0,0 +1,131 @@
+@Library('xmos_jenkins_shared_library@v0.27.0')
+
+def runningOn(machine) {
+  println "Stage running on:"
+  println machine
+}
+
+getApproval()
+pipeline {
+  agent none
+
+  parameters {
+    string(
+      name: 'TOOLS_VERSION',
+      defaultValue: '15.2.1',
+      description: 'The XTC tools version'
+    )
+    booleanParam(
+      name: 'XMATH_SMOKE_TEST',
+      defaultValue: true,
+      description: 'Enable smoke run'
+    )
+  } // parameters
+  options {
+    skipDefaultCheckout()
+    timestamps()
+    buildDiscarder(xmosDiscardBuildSettings())
+  } // options
+
+  stages {
+    stage('Bullds and tests') {
+      parallel {
+        stage('Linux builds and tests') {
```

Comment:
why don't we parallelize the xs3 tests with the x86 ones? We will reduce the duration of the Jenkins run by a considerable amount.

Reply:
The way it's designed right now is that the smoke run takes 4 min, and the non-smoke run takes 30 min. This whole stage is executed on a single agent. It's not gonna bring much benefit for a smoke run. and will reduce a non-smoke run by 4 min as xs3 takes 27 min and x86 takes 4 min.

Reply:
Thanks for the explanation and the details